src/java.base/share/classes/java/lang/classfile/attribute/SignatureAttribute.java

```diff
@@ -107,6 +107,8 @@ default Signature asTypeSignature() {
      * {@return a {@code Signature} attribute for a class}
      *
      * @param classSignature the class signature
+     * @throws IllegalArgumentException if the raw signature string is not
+     *         representable by a {@link Utf8Entry}
```

Comment:
I think adding this paragraph potentially to every place where a String is converted to Utf8Entry is polluting the javadoc. General paragraph about throwing IAE from Class-File API should cover that situations. We are not shadowing the specs here, we just need to say an IAE may happen if the class file is not possible to construct for any reason.